src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -639,7 +651,10 @@ def _db_dw_create(
 
     # If sku.name is not specified, resolve the requested sku name
     # using capabilities.
-    kwargs['sku'] = _find_db_sku_from_capabilities(cli_ctx, kwargs['location'], sku)
+    kwargs['sku'] = _find_db_sku_from_capabilities(cli_ctx, 
+        kwargs['location'], 
+        sku, 
+        requesting_serverless=(kwargs['compute_model'] == "Serverless"))
```

Comment:
Strings should not be hardcoded. If needed (which I hope not, because you should just compare capability's compute model with requested compute model), then use enum from SDK

Reply:
I do not think we have a compute model in the capabilities, we have a different SLO name for serverless, but we did not introduce a new compute model just for it.

Reply:
This is a problem. CLI (or any other client) really should discover capabilities with capabilities API. Parsing SKU name is not maintainable.

Reply:
Agreed, but I think it was decided in some other design. Powershell is doing the same for this as well. Maybe we should add an item for repair?

Reply:
Sorry, I will not knowingly accept new technical debt into CLI.

Reply:
Understood, and somewhat agree. We might need to discuss with the stakeholders on this issue.

Reply:
Avoiding this will introduce further bugs, so this is the best option for now until capabilities api is improved. I would like to prioritise fixing this debt asap.